src/hotspot/share/gc/shared/genCollectedHeap.cpp

```diff
@@ -853,6 +831,15 @@ void GenCollectedHeap::collect(GCCause::Cause cause, GenerationType max_generati
     return;
   }
 
+  bool should_run_young_gc =  (cause == GCCause::_wb_young_gc)
+                           || (cause == GCCause::_gc_locker)
+                DEBUG_ONLY(|| (cause == GCCause::_scavenge_alot))
+                           ;
+
+  const GenerationType max_generation = should_run_young_gc
+                                      ? YoungGen
+                                      : OldGen;
+
```

Comment:
At high risk of sounding onerous and nitpicky: please avoid adding new ways of formatting the `?` statement, and in this case it's short enough to write on a single line without any loss of clarity too.

(Fwiw, I searched for multi-line formatting of `?` in shared code, and as far as I can see the `?` isn't aligned to the `=` above ever - it is seemingly in the majority of times intended like a new scope, i.e. 2 (or sometimes 4?) spaces from the previous line. Actually I wanted to avoid these kinds of discussions, but, well it kind of pops out when looking at it...)

Reply:
I can see one such example in `jvm_posix.cpp`. Anyway, I feel this is clearer. (Very subjective ofc.)